differentiate unread counters

mahibi opened the pull request:
resolve #1559 

differentiate unread counters for group mentions and direct mentions

before | after (server with new capability) | after (server without new capability)
------- | -------- | ---------- 
![grafik](https://user-images.githubusercontent.com/2932790/134313062-1e6fd14e-e8c8-4798-b38d-7e3aa1664ed4.png) | ![grafik](https://user-images.githubusercontent.com/2932790/134365547-f3627d75-c0f5-409b-bda6-aa9ab0360733.png) | ![grafik](https://user-images.githubusercontent.com/2932790/134318104-5531cd27-06f4-437d-bab9-305a03bc961b.png) | 




Signed-off-by: Marcel Hibbe <dev@mhibbe.de>

nickvergessen commented on app/src/main/java/com/nextcloud/talk/adapters/items/ConversationItem.java:
```diff
@@ -136,20 +137,42 @@ public void bindViewHolder(FlexibleAdapter<IFlexible> adapter, ConversationItemV
+                    } else {
+                        holder.dialogUnreadBubble.setChipBackgroundColor(lightBubbleFillColor);
+                        holder.dialogUnreadBubble.setTextColor(lightBubbleTextColor);
+                        holder.dialogUnreadBubble.setChipStrokeWidth(4.0f);
```
From the screenshots I think the border could be thicker.

nickvergessen replied:
@marcoambrosini can you have a look at the screenshots and what border size is the web UI going to use?

mahibi replied:
stroke width 4.0
![grafik](https://user-images.githubusercontent.com/2932790/134319657-a623eee8-4f38-4013-af87-d5ee56d35773.png)

stroke width 5.0
![grafik](https://user-images.githubusercontent.com/2932790/134320206-07a1658d-9892-469e-967c-d7319d7225da.png)

stroke width 6.0
![grafik](https://user-images.githubusercontent.com/2932790/134319951-09d944ce-32e1-485f-979f-132bf31dd275.png)

mahibi replied:
updated screenshots. now with stroke width 6.0

nickvergessen reviewed (approved):
Looks good technical, only the border size is to discuss I think.

marcoambrosini reviewed (changes requested):
@marcelklehr could you please match the style of the new counter to the one we have in the vue library? 
<img width="270" alt="Screenshot 2021-09-07 at 10 05 04" src="https://user-images.githubusercontent.com/26852655/132308175-723d1bcf-03bd-4667-b958-f3f71e05f3a2.png">

mahibi commented:
> @marcelklehr could you please match the style of the new counter to the one we have in the vue library?
> <img alt="Screenshot 2021-09-07 at 10 05 04" width="270" src="https://user-images.githubusercontent.com/26852655/132308175-723d1bcf-03bd-4667-b958-f3f71e05f3a2.png">

changed background for group mentions to white and updated screenshots @marcoambrosini

jancborchardt commented:
> changed background for group mentions to white and updated screenshots @marcoambrosini

2 more details there, as shown in the screenshot @marcoambrosini posted:
- When the border is blue (team mention only), the number should also be blue
- The text needs more padding, that is the bubble size should increase a bit – ideally make it look exactly like on server :)

mahibi commented:
>     * When the border is blue (team mention only), the number should also be blue
done

>     * The text needs more padding, that is the bubble size should increase a bit – ideally make it look exactly like on server :)

i played around with that... it's possible to change the padding, but depending on which number is inside the bubble this looks quite bad. For example a "31" looks quite okay, but a 12 feels like there is too much padding at the beginning. i saw that @AndyScherzinger already set the current values startPadding=-3dp and endPadding=-1dp and i can confirm these values are the best compromise.

as an example see the following screenshots with padding "0":
![grafik](https://user-images.githubusercontent.com/2932790/134364268-adbc5268-1b55-49e7-a6a1-6021056a6bf8.png)

jancborchardt commented:
> @AndyScherzinger already set the current values startPadding=-3dp and endPadding=-1dp and i can confirm these values are the best compromise.

Reducing the left padding sounds good, but the right padding seems ok at 0. And there should be a bit more padding on top and bottom. :)

nickvergessen commented:
The top/bottom padding is clashing with the timestamp (which should be shown like in any other messenger) which is not there on the web.

mahibi commented:
discussed with @jancborchardt in person:
- left & right padding values are best like they were before.
- top & bottom padding won't change (and can't be changed that easily)
- date won't be bold anymore

Ivansss commented:
> changed background for group mentions to white and updated screenshots

Should the background be white or same color as the cell background?

![IMG_E0093](https://user-images.githubusercontent.com/4638605/134701796-453d9a1e-bafc-4e6c-afaf-484f8fc9d770.JPG)

@marcoambrosini @jancborchardt

nickvergessen commented:
Background color, otherwise the fallback of primary element will fail
